File: miniMaxSum.ts
export const miniMaxSum = (arr: number[] = []): [number, number] => {
  // Throw error if the array length is not equal to 5
  if (arr.length !== 5) throw new Error('Array length must be equal to 5')

  // Initialize the minSum and maxSum variables
  let minSum: number = Number.POSITIVE_INFINITY
  let maxSum: number = Number.NEGATIVE_INFINITY

  // Find the total sum of the numbers in the array
  const totalSum: number = arr.reduce((sum, num) => sum + num)

  // Iterate through the array
  for (const num of arr) {
    // Find the current sum
    const currentSum: number = totalSum - num

    // Compare the current sum with the min and max sums
    minSum = Math.min(minSum, currentSum)
    maxSum = Math.max(maxSum, currentSum)
  }

  // Return the min and max sums
  return [minSum, maxSum]
}

File: miniMaxSum.py
# Source: https://www.hackerrank.com/challenges/mini-max-sum/problem?isFullScreen=true
def miniMaxSum(arr):
    # Initialize the min and max sums
    min_sum, max_sum = float("inf"), float("-inf")

    # Find the sum of all numbers in the array
    total_sum = sum(arr)

    # Loop through the array, subtract the current number for each step
    for num in arr:
        # Obtain the current sum
        current_sum = total_sum - num

        # Compare the current sum with the min and max sums
        if current_sum < min_sum:
            min_sum = current_sum
        if current_sum > max_sum:
            max_sum = current_sum

    # Return the min and max sums
    return [min_sum, max_sum]


print(miniMaxSum([1, 2, 3, 4, 5]))
